tests/metal/nested-struct-fragment-output.slang

```diff
@@ -0,0 +1,40 @@
+//TEST:SIMPLE(filecheck=METAL): -target metal -stage fragment -entry fragmentMain
+//TEST:SIMPLE(filecheck=METALLIB): -target metallib -stage fragment -entry fragmentMain
+
+//METAL-DAG: color(0)
+//METAL-DAG: color(1)
+//METAL-DAG: color(2)
+//METALLIB: @fragmentMain
+
+RWStructuredBuffer<float> outputBuffer;
+
+struct NestedFragment
+{
+    float2 p3;
+};
+struct Fragment
+{
+	float4 p1;
+    float3 p2;
+    NestedFragment p3;
```

Comment:
Add one more NestedFragment field here.

Reply:
Also need to test explicit semantics.
Does it work if we have a nested struct with explicit semantics, but no explicit semantics inside the nested struct fields?